linera-core/src/chain_worker/state/attempted_changes.rs

```diff
@@ -363,6 +356,14 @@ where
             .storage
             .maybe_write_blob_states(&blob_ids, blob_state)
             .await?;
```

Comment:
Everything up to here could probably also be deduplicated.

I think if we check `tip.next_block_height < height` in just two places (or even just one?), we can remove `preprocess_certificate` entirely.